docs/source/common/hyperparameters.rst

```diff
@@ -113,78 +113,75 @@ LightningModule hyperparameters
-                # call this to save (layer_1_dim=128, learning_rate=1e-4) to the checkpoint
-                self.save_hyperparameters()
+.. code-block:: python
 
-                # equivalent
-                self.save_hyperparameters("layer_1_dim", "learning_rate")
+    class LitMNIST(LightningModule):
+        def __init__(self, layer_1_dim=128, learning_rate=1e-2):
+            super().__init__()
+            # call this to save (layer_1_dim=128, learning_rate=1e-4) to the checkpoint
+            self.save_hyperparameters()
 
-                # Now possible to access layer_1_dim from hparams
-                self.hparams.layer_1_dim
+            # equivalent
+            self.save_hyperparameters("layer_1_dim", "learning_rate")
 
+            # Now possible to access layer_1_dim from hparams
+            self.hparams.layer_1_dim
 
-2.  Sometimes your init might have objects or other parameters you might not want to save.
-    In that case, choose only a few
 
-    .. code-block:: python
+In addition, loggers that support it will automatically log the contents of ``self.hparams``.
 
-        class LitMNIST(LightningModule):
-            def __init__(self, loss_fx, generator_network, layer_1_dim=128 ** kwargs):
-                super().__init__()
-                self.layer_1_dim = layer_1_dim
-                self.loss_fx = loss_fx
+Excluding hyperparameters
+"""""""""""""""""""""""""
 
-                # call this to save (layer_1_dim=128) to the checkpoint
-                self.save_hyperparameters("layer_1_dim")
+By default, every parameter of the ``__init__`` method will be considered a hyperparameter to the LightningModule.
+However, sometimes some parameters need to be excluded from saving, for example when they are not serializable.
+In this case, exclude them explicitly:
 
+.. code-block:: python
 
-        # to load specify the other args
-        model = LitMNIST.load_from_checkpoint(PATH, loss_fx=torch.nn.SomeOtherLoss, generator_network=MyGenerator())
+    class LitMNIST(LightningModule):
+        def __init__(self, loss_fx, generator_network, layer_1_dim=128):
+            super().__init__()
+            self.layer_1_dim = layer_1_dim
+            self.loss_fx = loss_fx
 
+            # call this to save only (layer_1_dim=128) to the checkpoint
+            self.save_hyperparameters("layer_1_dim")
 
-3.  You can also convert full objects such as ``dict`` or ``Namespace`` to ``hparams`` so they get saved to the
-    checkpoint.
+            # equivalent
+            self.save_hyperparameters(exclude=["loss_fx", "generator_network"])
 
-    .. code-block:: python
 
-        class LitMNIST(LightningModule):
-            def __init__(self, conf: Optional[Union[Dict, Namespace, DictConfig]] = None, **kwargs):
-                super().__init__()
-                # save the config and any extra arguments
-                self.save_hyperparameters(conf)
-                self.save_hyperparameters(kwargs)
```

Comment:
I understand you don't want to document this as it can be a footgun for users.

But in that case, should also deprecate this behaviour? What's the point of having support but keep it undocumented?

Reply:
It is still documented here in the method signature: https://github.com/PyTorchLightning/pytorch-lightning/blob/a9bd4fbd96c1e73e251859d99b207008008de87d/pytorch_lightning/core/mixins/hparams_mixin.py#L33

I'm also ok going for a deprecation, but I should create an issue so we can discuss it there. Nonetheless, I think it's better to go with this documentation change regardless.

Reply:
Sure

Reply:
Created #10375 for everyone to see :)